client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
 	} );
 
 	test( 'should render Card if dismissPreferenceName is null', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ null } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } dismissPreferenceName={ null } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).not.toHaveClass( 'is-dismissible' );
 	} );
 
 	test( 'should render DismissibleCard if dismissPreferenceName is defined', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
-		expect( comp.find( Card ) ).toHaveLength( 0 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 1 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
+		expect( screen.getByLabelText( 'Dismiss' ) ).toBeInTheDocument();
 	} );
 
 	test( 'should have .has-call-to-action class if callToAction is defined', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
+		expect( container.firstChild ).toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should not have .has-call-to-action class if callToAction is null', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ null } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } callToAction={ null } /> );
+		expect( container.firstChild ).not.toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should render a <Button /> when callToAction is specified', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Buy something!' } /> );
-		expect( comp.find( Button ) ).toHaveLength( 1 );
+		render( <Banner { ...props } callToAction={ 'Buy something!' } /> );
+		expect( screen.getByRole( 'button' ) ).toHaveTextContent( 'Buy something!' );
 	} );
 
 	test( 'should not render a <Button /> when callToAction is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( Button ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByRole( 'button' ) ).toBeNull();
 	} );
 
 	test( 'should have .is-jetpack class and JetpackLogo if jetpack prop is defined', () => {
 		const { plan, ...propsWithoutPlan } = props;
-		const comp = shallow( <Banner { ...propsWithoutPlan } jetpack /> );
-		expect( comp.find( '.is-jetpack' ) ).toHaveLength( 1 );
-		expect( comp.find( 'JetpackLogo' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...propsWithoutPlan } jetpack /> );
+		expect( container.firstChild ).toHaveClass( 'is-jetpack' );
+		expect( container.getElementsByTagName( 'svg' ) ).toHaveLength( 1 );
```

Comment:
I was reading that [when using an SVG it is a good idea to use `role="img"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role#svg_and_roleimg). Do you think it would be worth adding this to the Gridicon and JetpackLogo components?

Reply:
Hm, it really depends. In my understanding, the `img` role is useful for content images, images that actually have a meaning and make sense in the context's content. That doesn't seem to be true for those logos, which often are there for just cosmetic purposes, and don't have much meaning when accessing the app with a screen reader. 

I guess it's a matter of interpretation. If you believe we should add an `img` role, I think we should also make sure to have an `alt` attribute as well.

A side note here: while `container.getElementsByTagName()` will do the job, it's one of the methods that directly performs a DOM query and those are generally discouraged. It's hard to notice that, because we haven't yet enabled the ESLint rules for `@testing-library/react`, but when we do that, it'd trigger an ESLint error. With that in mind, we could consider adding a `role="img"`, to those logos, but that could be its own PR, since it may appear in many places throughout the app.

So to summarize: adding a `role="img"` here with the appropriate `alt` makes the most sense in the long run IMHO.

Reply:
Ah, okay! I was thinking because of this statement via the link I shared above that it might be nicer on screen readers, so it doesn't read out the child nodes: 

> "If you are using embedded SVG images in your page, it is a good idea to set role="img" on the outer <svg> element and give it a label. This will cause screen readers to just consider it as a single entity and describe it using the label, rather than trying to read out all the child nodes"

But I'm not too familiar with screen readers yet, so I could be misunderstanding. 😅 

In regards to your site note, I originally had `expect( container.getElementsByClassName( 'jetpack-logo' ) ).toHaveLength( 1 );`

Is it better to look for the class name instead? Wouldn't that perform a query on the DOM as well, though? I saw the following in a test and thought it might be a preferred approach: 

https://github.com/Automattic/wp-calypso/blob/8b56c977bd83719a89628f34ee6e09f62995b6a7/client/components/external-link/test/index.js#L28

I'm now guessing it was done here since there isn't a specific class for the SVG (aside from generic Gridicon ones)?

Reply:
Yeah, the thing is that there aren't many good ways to locate those components if we don't perform a DOM query. That's why we went for `getElementsByTagName` in that previous test - IMHO there also isn't any suitable role we could provide to that icon that only has a presentational purpose.

On the other hand, the svg-s in the logo components could have the `img` role with an `alt` stating what kind of logo it is, as I suggested above. 

If you're not happy with that idea, feel free to keep what we have. We already have a bunch of those instances and we might have to perform some additional refactoring if we decide to enable those ESLint rules I've mentioned above.